### PR TITLE
Fix for lack of url_id in viewing operation

### PR DESCRIPTION
**Pull request by stefanoborini**
Adds a label to the container with the url_id, so that we can recover it.

### CODE REVIEW

**Comment by stefanoborini** [comment]
Fixes #43

**Comment by kitchoi** [comment]
Just tested that `url_id` still cannot be obtained.  The change here is based on the API of `docker.client.inspect_container` while `HomeHandler._container_from_options` uses `docker.containers` which returns dictionaries in a different structure.o

"Labels" is available as `docker_dict['Labels']` for the results of `docker.containers`.  But the same labels are available under `docker_dict['Config']['Labels']` for results of `docker.inspect_container`.

**Comment by kitchoi** [comment]
@stefanoborini My bad, I should loop over traits_names in here: https://github.com/simphony/simphony-remote/blob/be89fe0f281af48666e6409ded4b00fe2c755482/tests/utils.py#L233

**Comment by stefanoborini** [comment]
@kitchoi I am going to fix the test.

**Comment by stefanoborini** [comment]
At this point I start to question the general use of defaults for those gets. If the labels are not there, it's a broken container. We shouldn't keep going.

File: tests/utils.py [hunk 1]
```diff
@@ -183,6 +184,46 @@ def mock_docker_client_with_existing_stopped_container():
...
...fbIwhCOBLp7fXt3LVr0aKFXq+35NqR118/5op+F6q09MZZs2adbnc+/cwzzc3Nfe7U5/e/++6fqqqq0tMznnzyyYFb9EAZDIann35aVuS9X/1dFMWBZqEocs6cOQiBNWvWrl37htVinTp1yqCo+i2f6urqpUuXlVx7bVFxMcuyLS0tn9fVeTwerVa3bNmLsVMj/tXDC/qEcyWDOQs0+PRHSPlh/ke5UDcQig0TOY77wx/+q7Oz88OqKqfTee+ce4uKijxez6effPrRRx/q9fqKiorCwsL+qxMNOiII4c0333xTWdn2zz4nSXLQCiOvvrqoqOjDjz5SFGXuLx5ITU29eDJOrVYnJCTu/WpvXX0dAICiaLPZNGXK1CcWLSgrK4vlrE/QGQxGnU7bVwgh1Gh1FotFp9MODKgsFovRZIp7qY0kSZPZZLEkMzF5KxWrslgs0Uz4ha1P03RycrKiKBQVn3QjIDQkJhoMier+WUWr1frnP/9pzZo1723+6zPPPEOSJEIKy7Jjx92w9Le/HTdubOwYCYIwGAxJSckcN8hPoPR6/XPPPXvypIMXBL1OOzDxZzQaZ8+++4UXXoCQnDp12sDc3CCX8qIodnV1OxwOt8cNAWAYJjs7OzMzk+O42J5FE58ulyslxZaaaovJhro6OlxWqzUtLS22Ps/zR48e1Wg0eXl5sWaNRCIOh4PnhZycYQkJCX1Xcg7HKbWay8vLoy+Y6w4EAk1NTQiB3NwctVrd/y5Famlp8Xp7srIy43Lp0U/b2ttPnDgRDAYJSKSnp+fl5Wq12jhPIopiU1OTz+fLysoa2Ei0nWPHjomiCCGRkzOsL4Hfp/r6L26dNGl4cdG2bdvOtwIAwro8UhRl8+a/AgDmz3+0L70xUPhvRVwu+Xy+999/n2XZu+6adYH7egpbaqhvkiVRFHt7e1etWvXJJ5/MmDFj9OgLXdb/lL+U//cTQuirr/6+bt1bzc0t/7N3b2FBQXX1R/n5+Rf5DtZQSZblDz74gKJorVY3adLkQ4cOxd2LDRReAT+x8CaMAWAAWBgABoCFAWAAWBgABoCFAWAAWBgABoCFAWAAWBgABoCFAWAAWBgABoCFAWAAWBgABoCFAWAAWBgABoCFAWAAWBgABoCFAWAAWBgABoCFAWAAWBjAlaf/BQGW9UX8qIqDAAAAAElFTkSuQmCC====',
```
Comment (kitchoi) [inline tests/utils.py]: The icon is not tested.  Can we not have the icon in the test case please?

**Comment by kitchoi** [comment]
Thanks @stefanoborini 
LGTM.

> At this point I start to question the general use of defaults for those gets. If the labels are not there, it's a broken container. We shouldn't keep going.

I tend to think of `Container.from_docker_containers_dict` to be a faithful adapter for parsing dict to a `Container` that does not care about the validating the content of a container.

File: remoteappmanager/docker/container.py [hunk 2]
```diff
@@ -47,14 +47,16 @@ def host_url(self):
...
-                'image_id={3}, ip={4}, port={5}>').format(
-                    self.docker_id, self.name,
-                    self.image_name, self.image_id,
-                    self.ip, self.port)
+        return (
+            '<Container(' +
+            ", ".join(
+                ["{}={}".format(name, getattr(self, name))
+                    for name in self.trait_names()
+                 ]) +
```
Comment (kitchoi) [inline remoteappmanager/docker/container.py]: one minor comment: no list creation is needed here.

**Comment by kitchoi** [comment]
There is one super minor comment that I would be happy to merge as is.  Please feel free to merge.

**Pull request merged**